### PR TITLE
Expose prometheus metrics

### PR DESCRIPTION
**Pull request by MaksTkachuk**

### CODE REVIEW

File: api_tests/gql_tests/timeseries.py [hunk 1]
```diff
@@ -80,6 +80,7 @@ def test_timeseries_metrics(slug_test_suite, timeseries_metrics, slug_progress,
                     test_case.response = result
 
         test_case.finished_at = dt.utcnow()
+        test_case.delay = (test_case.finished_at - delayed_since).total_seconds()
```
Comment (kalkov) [inline api_tests/gql_tests/timeseries.py:83]: What does delayed_since represent?

Comment (kalkov) [inline api_tests/gql_tests/timeseries.py:83]: What does delayed_since represent?

File: app.py [hunk 2]
```diff
@@ -217,3 +225,18 @@ def _elapsed_time_category(elapsed_time):
         return "medium"
     else:
         return "slow"
+
+
+def generate_metrics(test_suite):
```
Comment (kalkov) [inline app.py:230]: rename to `generate_prometheus_metrics`

File: app.py [hunk 3]
```diff
@@ -217,3 +225,18 @@ def _elapsed_time_category(elapsed_time):
...
+    data = test_suite.output_for_html()
+    lines = []
+    for project in data:
+        for item in project['data']:
+            line_status = f'api_test_status{{asset="{project["slug"]}", metric="{item["name"]}"}} {item["status"]}'
```
Comment (kalkov) [inline app.py:235]: `api_test_status` -> `api_tests_status`

File: app.py [hunk 4]
```diff
@@ -217,3 +225,18 @@ def _elapsed_time_category(elapsed_time):
...
+            line_delay = f'api_test_delay{{asset="{project["slug"]}", metric="{item["name"]}"}} {item["delay"]}'
```
Comment (kalkov) [inline app.py:236]: `api_test_delay` -> `api_tests_delay`

File: app.py [hunk 5]
```diff
@@ -217,3 +225,18 @@ def _elapsed_time_category(elapsed_time):
...
+            line_performance = f'api_test_query_duration{{asset="{project["slug"]}", metric="{item["name"]}"}} {item["elapsed_time"]}'
```
Comment (kalkov) [inline app.py:237]: `api_test_query_duration` -> `api_tests_query_duration`

File: app.py [hunk 6]
```diff
@@ -217,3 +225,18 @@ def _elapsed_time_category(elapsed_time):
...
+            lines.append(line_status)
```
Comment (kalkov) [inline app.py:238]: Also add `TYPE` and `HELP` lines. Example:
```
# HELP airflow_dag_status Shows the number of dag starts with this status
# TYPE airflow_dag_status gauge
airflow_dag_status{dag_id="airflow-clear-missing-dags",owner="operations",status="failed"} 4.0
airflow_dag_status{dag_id="airflow-clear-missing-dags",owner="operations",status="success"} 49.0
airflow_dag_status{dag_id="airflow-failing",owner="vassil.k@santiment.net",owner_discord="@381663669303377921",status="failed"} 2.0
airflow_dag_status{dag_id="anomalies-daily-historical",owner="vlad.v@santiment.net",owner_discord="@191504103975813120",status="failed"} 1.0
```

File: app.py (hunk 3 above)
Comment (kalkov) [reply app.py:235]: Are you sure string is a proper type for value in prometheus?
```
api_test_status{asset="bitcoin", metric="stock_to_flow"} GraphQL error
api_test_status{asset="bitcoin", metric="supply_on_exchanges"} corrupted
api_test_status{asset="bitcoin", metric="supply_outside_exchanges"} corrupted
api_test_status{asset="bitcoin", metric="total_supply"} GraphQL error
```

File: api_tests/gql_tests/timeseries.py (hunk 1 above)
Comment (MaksTkachuk) [reply api_tests/gql_tests/timeseries.py:83]: delayed_since is the diff between time of the test and the last available data point

File: app.py (hunk 2 above)
Comment (MaksTkachuk) [reply app.py]: done

File: app.py (hunk 4 above)
Comment (MaksTkachuk) [reply app.py]: done

File: app.py (hunk 5 above)
Comment (MaksTkachuk) [reply app.py]: done

File: app.py [hunk 7]
```diff
@@ -217,3 +225,29 @@ def _elapsed_time_category(elapsed_time):
...
+
+
+def generate_prometheus_metrics(test_suite):
+    data = test_suite.output_for_html()
+    lines_status = [
+        '# HELP api_tests_status Status of the metric after the last run: 0 - passed, 1 - corrupted, 2 - empty, 3 - GraphQL error',
+        '# TYPE api_tests_status gauge'
+    ]
+    lines_delay = [
+        '# HELP api_tests_delay Time delta between the test run and last available data point in seconds',
```
Comment (kalkov) [inline app.py:237]: between the latest test run

File: app.py [hunk 8]
```diff
@@ -217,3 +225,29 @@ def _elapsed_time_category(elapsed_time):
         return "medium"
     else:
         return "slow"
+
+
+def generate_prometheus_metrics(test_suite):
+    data = test_suite.output_for_html()
+    lines_status = [
+        '# HELP api_tests_status Status of the metric after the last run: 0 - passed, 1 - corrupted, 2 - empty, 3 - GraphQL error',
```
Comment (kalkov) [inline app.py:233]: from the latest test run

File: app.py (hunk 7 above)
Comment (kalkov) [inline app.py:237]: between the latest test run

**Review by kalkov — APPROVED** [review]

**Pull request merged**